### PR TITLE
Add GeoRadiusByMember support

### PR DESCRIPTION
**Pull request by azuredream**
#1548

### CODE REVIEW

File: src/server/zset_family.cc [hunk 1]
```diff
@@ -2762,34 +2765,63 @@ void GeoSearchGeneric(ConnectionContext* cntx, const GeoShape& shape_ref, string
...
+    auto store_cb = [&](Transaction* t, EngineShard* shard) {
+      if (shard->shard_id() == dest_shard) {
+        ZParams zparams;
+        zparams.override = true;
+        add_result = OpAdd(t->GetOpArgs(shard), zparams, store_key, ScoredMemberSpan{smvec}).value();
+      }
+      return OpStatus::OK;
+    };
+    cntx->transaction->Schedule();
+    cntx->transaction->Execute(std::move(store_cb), true);
```
Comment (azuredream) [inline src/server/zset_family.cc:2820]: I'm not sure which one should I use here.  `schedule-execute` or `ScheduleSingleHopT`? It seems that I have to set "-4, 3, 3, 1" in `Register` to use  `schedule-execute`. But got error message:
_127.0.0.1:6379> GEORADIUSBYMEMBER Europe Madrid 700 KM WITHCOORD WITHDIST STORE test1
Error: Server closed the connection_

The search part works good. 

I'd appreciate it if anyone could give me some advice.

Comment (romange) [reply src/server/zset_family.cc:2820]: Please do not add support for `STORE` and `STOREDIST` options. There are several reasons for this:
1. it requires some adjustments to transactional framework to support this non-standard command layout
2. this also requires advanced knowledge of how to build such commands. 
 
Having said that, implementing all the other options will  provide substantiation contribution and will make it easier for us to close the gaps.

Comment (azuredream) [reply src/server/zset_family.cc]: Thanks for your suggestion. I've updated my PR.
 
I'll be glad to make further changes to transactional framework if you could give me some suggestions.

Another thing on my to-do list is that GeoSearch fetches redundant data and subsequently filter them. To eliminate the intermediate result array, also requires some change to the standard search process which cannot take extra geo filter as input.

**Review by romange — APPROVED** [review]

**Comment by romange** [comment]
@azuredream if you want, please add disabled unit tests for the missing features   (see `GTEST_SKIP()`).
This way I will be able to see what needs to be done for the test to pass and I will be able to guide you.

**Pull request merged**